apps/web/src/components/activity/ExecutionDetailShowRaw.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from 'react';
+import { useMantineColorScheme } from '@mantine/core';
+import { Prism } from '@mantine/prism';
+import styled from 'styled-components';
+
+import { Button, colors, Text } from '../../design-system';
+
+const DetailTitle = styled(Text)`
+  font-size: 14px;
+  font-weight: 700;
+  line-height: 17px;
+  padding-bottom: 20px;
+`;
+
+const ActionButton = styled(Button)<{ theme: string }>`
+  background-color: transparent;
+  background-image: none;
+  border: 1px solid ${({ theme }) => (theme.colorScheme === 'dark' ? colors.B40 : colors.B80)};
+  border-radius: 7px;
+  box-shadow: none;
+  color: ${({ theme }) => (theme.colorScheme === 'dark' ? colors.white : colors.B40)};
+  font-size: 12px;
+  font-weight: 700;
+  height: 30px;
+  padding: 0;
+  width: 95px;
+`;
+
+export const ExecutionDetailShowRaw = ({ raw, onShowExecutionDetail, onHideExecutionDetail }) => {
+  const theme = useMantineColorScheme();
+  const [showSnippet, setShowSnippet] = useState<boolean>(false);
+
+  const action = showSnippet ? onHideExecutionDetail : onShowExecutionDetail;
+  const label = showSnippet ? 'Close detail' : 'Show detail';
```

Comment:
Maybe we should add an x button to close this? And not only toggle with the button?